zds/tutorialv2/tests/tests_views/tests_editlabels.py

```diff
@@ -0,0 +1,114 @@
+        """As login is required, unauthenticated users shall be redirected to the login page."""
+        self.client.logout()
+        response = self.client.post(self.good_url)
+        self.login_url = f"{reverse('member-login')}?next={self.good_url}"
+        self.assertRedirects(response, self.login_url)
+
+    def test_simple_user_forbidden(self):
+        """Simple users shall not be able to access to the view."""
+        self.client.force_login(self.user)
+        response = self.client.post(self.good_url)
+        self.assertEqual(response.status_code, 403)
+
+    def test_staff_authorized(self):
+        """Staff shall have access to the view."""
+        self.client.force_login(self.staff)
+        response = self.client.post(self.good_url)
+        self.assertRedirects(response, self.success_url)
+
+
+class EditLabelsFunctionalTests(TestCase):
+    def setUp(self):
+        self.staff = StaffProfileFactory().user
+        self.content = PublishableContentFactory()
+        self.content = PublishableContentFactory()
+        self.url = reverse("content:edit-labels", kwargs={"pk": self.content.pk})
+        self.labels = [LabelFactory() for _ in range(3)]
+
+    @patch("zds.tutorialv2.signals.labels_management")
+    def test_labels_updated(self, labels_management):
+        self.client.force_login(self.staff)
+        response = self.client.post(self.url, {"labels": [label.pk for label in self.labels]}, follow=True)
+        self.assertEqual(list(self.content.labels.all()), self.labels)
+        self.assertContains(response, "alert-box success")
+        self.assertEqual(labels_management.send.call_count, 1)
+
+    @patch("zds.tutorialv2.signals.labels_management")
+    def test_invalid_parameters(self, labels_management):
+        self.client.force_login(self.staff)
+        response = self.client.post(self.url, {"labels": [42]}, follow=True)
+        self.assertEqual(list(self.content.labels.all()), [])
+        self.assertContains(response, "alert-box alert")
+        self.assertFalse(labels_management.send.called)
+
+
+class AssignLabelsTestCase(TestCase):
+    def setUp(self):
+        self.staff = StaffProfileFactory().user
+        self.content_with_label = PublishedContentFactory()
+        self.content_without_label = PublishedContentFactory()
+        self.url = reverse("content:edit-labels", kwargs={"pk": self.content_with_label.pk})
+        self.label = LabelFactory()
+
+    def test_add_label(self):
+        self.client.force_login(self.staff)
+        self.client.post(self.url, {"labels": [self.label.pk]}, follow=True)
+        self.assertIn(self.label, self.content_with_label.labels.all())
+        self.assertNotIn(self.label, self.content_without_label.labels.all())
+
+    def test_remove_label(self):
+        self.client.force_login(self.staff)
+        self.client.post(self.url, {"labels": [self.label.pk]}, follow=True)
+        self.content_with_label.labels.remove(self.label)
```

Comment:
Là vous appelez la vue pour ajouter un label, puis vous supprimez le label avec un mécanisme de Django (qui donc doit être testé chez Django). C'est l'inverse qu'il faut faire :
```suggestion
        self.content_with_label.labels.add(self.label)
        self.client.post(self.url, {"labels": []}, follow=True)
```